.github/workflows/doc-gen.yml

```diff
@@ -0,0 +1,34 @@
+name: DocGenCI
+
+on:
+  workflow_run:
+    workflows: [CI]
+    types: [completed]
+
+permissions:
+  id-token: write
+
+jobs:
+  stable-release:
+    if: github.event.workflow_run.conclusion == 'success' && github.repository == 'aws/karpenter'
```

Comment:
Any reason it should just run on completion? Just to reduce the number of API calls?